### PR TITLE
PYIC-5953: Clean up session credential table switch

### PR DESCRIPTION
**Pull request by Wynndow**
**Not to be merged until https://github.com/govuk-one-login/ipv-core-common-infra/pull/906, and we're happy with the changes running in prod**

### What changed

Clean up session credential table switch

### Why did it change

This should be merged once the reads from the sessions credenentials table are happening in prod, and we're happy. It does a few things:

* Removes use of the sessionCredentialsTableReads feature flag
* Stops dual writing VCs. This stops writing to the user-issued-credentials-table during a journey. It's now only written to at the end of a journey
* Removes now unneccesary AWS perms from some lambdas
* Removes some now unncessary filtering from the "working" set of VCs. We should be operating on VCs of a know profile type when creating shared attributes and building identities

### Issue tracking

- [PYIC-5953](https://govukverify.atlassian.net/browse/PYIC-5953)

[PYIC-5954]: https://govukverify.atlassian.net/browse/PYIC-5954?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PYIC-5953]: https://govukverify.atlassian.net/browse/PYIC-5953?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Joe-Edwards-GDS — COMMENTED** [review]
Looks reasonable, but won't approve until we're ready to merge.

Do you think we should also remove the feature flags from `CoreFeatureFlag.java`?

**Review by Joe-Edwards-GDS — DISMISSED** [review]

**Review by Joe-Edwards-GDS — DISMISSED** [review]

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/IpvSessionItem.java
Comment (Joe-Edwards-GDS) [inline libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/IpvSessionItem.java:42]: I think this is fine for now - we might need to rethink this entirely once we introduce EVCS though, see notes on https://govukverify.atlassian.net/browse/PYIC-6083

**Review by Joe-Edwards-GDS — APPROVED** [review]

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/IpvSessionItem.java
Comment (Joe-Edwards-GDS) [reply libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/IpvSessionItem.java:42]: (It's also worth checking this won't break any existing sessions, but I think it should be fine...)

Comment (Wynndow) [reply libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/IpvSessionItem.java:42]: I figured that as we're not live with inherited identities this should be fine. The reads and writes to this are with behind a flag, or only read if the user has an operation profile.

**Pull request merged**
